src/workloads/sharding/ReshardCollectionWithIndexes.yml

```diff
@@ -0,0 +1,174 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/sharding"
+Description: |
+  This test expects the target cluster is created using ebs snapshot with 1 billion 1KB records 
+  and has 3 shards, named shard-00, shard-01, shard-02. The collection has 10 indexes including 
+  _id index, single-key index and compound index. The whole dataset is on 1 shard at the beginning.
+  ReshardCollection should use same-key resharding and use shardDistribution parameter to reshard 
+  into 2 shards then 3 shards. There will be random CRUD operations during resharding but should 
+  run at a very low rate.
+
+  The workload consists of 3 phases:
+    1. Turning off balancer and make the sharded collection exists on only 1 shard.
+    4. Running read and write operations on the collection while it is being resharded to 2 shards.
+    5. Running read and write operations on the collection while it is being resharded to 3 shards.
+    
+
+  The inserted documents have the following form:
+
+      {
+        _id: integer(default _id index),
```

Comment:
All documents are generated through genny's data loader where the integer and length of short string fields are randomly generated. The main reasons I make it in this format are:

- We want every document is around 1KB
- We want to have multiple fields so we can form different combinations of indexes.

Regarding the type of those fields, I don't think that's important because we just want to see the performance improvements over the old way on a certain setup. The performance on different kinds of fields are not what we are looking to check here.

Reply:
Could you add some of the above to the comment string? Capturing that workload intent can be very valuable when we come back to the workload in the future.